spec/unit/network/authconfig_spec.rb

```diff
@@ -107,3 +107,29 @@
     end
   end
 end
+
+describe Puppet::Network::AuthConfig do
+  class TestAuthProvider
+    def initialize(rights=nil); end
+    def check_authorization(method, path, params); end
+  end
+
+  it "instantiates authprovider_class with rights" do
+    Puppet::Network::AuthConfig.authprovider_class = TestAuthProvider
+    rights = Puppet::Network::Rights.new
+    TestAuthProvider.expects(:new).with(rights)
+    described_class.new(rights)
+  end
+
+  it "delegates authorization check to authprovider_class" do
+    Puppet::Network::AuthConfig.authprovider_class = TestAuthProvider
+    TestAuthProvider.any_instance.expects(:check_authorization).with(:save, '/path/to/resource', {})
+    described_class.new.check_authorization(:save, '/path/to/resource', {})
+  end
+
+  it "uses DefaultAuthProvider by default" do
+    Puppet::Network::AuthConfig.authprovider_class = nil
```

Comment:
Ha, actually...while writing my comment above I realized I could probably reorder the tests so that I ensure the default class is used first before testing that I can set the class.

Although, one advantage of the way `AuthConfig` is now is that it's impossible to have a `nil` authprovider accidentally.

Happy to try to reorder these tests and them change/simplify AuthConfig to have `@@authprovider_class = DefaultAuthProvider`.